xonsh/tools.py

```diff
@@ -228,6 +228,30 @@ def __radd__(self, other):
             other = other._l
         return EnvPath(other + self._l)
 
+    def add(self, data, toStart=False, replace=False):
+        """
+        add a value to this EnvPath,
+        kwargs:
+            toStart:
+                Value will be added to start if True, or to the end
+                if it's False
+                Default: False
+            replace:
+                If True, the value will be removed and added to the
+                start or end(depending on the value of toStart)
+                Default: False
+        """
+        try:
+            i = self._l.index(data)
```

Comment:
Actually, you aren't even using `i` in the function, so this whole try except block can just be `if data in self._l`